features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/rolesandpermissions/permissions/ChangeRoomPermissionsView.kt

```diff
@@ -0,0 +1,179 @@
+                ChangeRoomPermissionsSection.MembershipModeration -> stringResource(R.string.screen_room_change_permissions_member_moderation)
+            }
+            TopAppBar(
+                title = { Text(text = title, style = ElementTheme.typography.aliasScreenTitle) },
+                navigationIcon = {
+                    BackButton(onClick = { state.eventSink(ChangeRoomPermissionsEvent.Exit) })
+                },
+                actions = {
+                    TextButton(
+                        text = stringResource(CommonStrings.action_save),
+                        onClick = { state.eventSink(ChangeRoomPermissionsEvent.Save) },
+                        enabled = state.hasChanges,
+                    )
+                }
+            )
+        }
+    ) { padding ->
+        Column(modifier = Modifier.padding(padding)) {
+            for ((index, permissionItem) in state.items.withIndex()) {
+                ListSectionHeader(titleForSection(item = permissionItem), hasDivider = index > 0)
+                SelectRoleItem(
+                    permissionsItem = permissionItem,
+                    role = RoomMember.Role.ADMIN,
+                    currentPermissions = state.currentPermissions
+                ) { item, role ->
+                    state.eventSink(ChangeRoomPermissionsEvent.ChangeRole(item, role))
+                }
+                SelectRoleItem(
+                    permissionsItem = permissionItem,
+                    role = RoomMember.Role.MODERATOR,
+                    currentPermissions = state.currentPermissions
+                ) { item, role ->
+                    state.eventSink(ChangeRoomPermissionsEvent.ChangeRole(item, role))
+                }
+                SelectRoleItem(
+                    permissionsItem = permissionItem,
+                    role = RoomMember.Role.USER,
+                    currentPermissions = state.currentPermissions
+                ) { item, role ->
+                    state.eventSink(ChangeRoomPermissionsEvent.ChangeRole(item, role))
+                }
+            }
+        }
+    }
+
+    AsyncActionView(
+        async = state.saveAction,
+        onSuccess = { state.eventSink(ChangeRoomPermissionsEvent.ResetPendingActions) },
+        onErrorDismiss = { state.eventSink(ChangeRoomPermissionsEvent.ResetPendingActions) }
+    )
+
+    AsyncActionView(
+        async = state.confirmExitAction,
+        onSuccess = { onBackPressed() },
+        confirmationDialog = {
+            ConfirmationDialog(
+                title = stringResource(R.string.screen_room_change_role_unsaved_changes_title),
+                content = stringResource(R.string.screen_room_change_role_unsaved_changes_description),
+                submitText = stringResource(CommonStrings.action_save),
+                cancelText = stringResource(CommonStrings.action_go_back),
```

Comment:
The problem with the third button is we don't have one in M3 by default, and the one we manually added doesn't work well with long texts for the actions. I agree the go back action can be a bit difficult to understand...

So it's not great, but I think we can leave it like this and come back to it later, because it needs design changes and that may take a while.